beets/test/helper.py

```diff
@@ -492,29 +604,99 @@ def load_plugins(self, *plugins: str) -> None:
-        plugins = (self.plugin,) if hasattr(self, "plugin") else plugins
-        self.config["plugins"] = plugins
+        if plugins:
+            self.config["plugins"] = plugins
+        elif self.plugin:
+            self.config["plugins"] = (self.plugin,)
+        else:
+            self.config["plugins"] = tuple()
         beets.plugins.load_plugins()
 
     def unload_plugins(self) -> None:
         """Unload all plugins and remove them from the configuration."""
         # FIXME this should eventually be handled by a plugin manager
         beets.plugins.BeetsPlugin.listeners.clear()
         beets.plugins.BeetsPlugin._raw_listeners.clear()
+        beets.plugins.BeetsPlugin.template_funcs.clear()
+        beets.plugins.BeetsPlugin.template_fields.clear()
+        beets.plugins.BeetsPlugin.album_template_fields.clear()
         self.config["plugins"] = []
         beets.plugins._instances.clear()
 
+    def get_plugin_instance(
+        self, name: str | None = None
+    ) -> beets.plugins.BeetsPlugin:
+        """Get the plugin instance for a registered and loaded plugin."""
+        name = name or self.plugin
+        for plugin in beets.plugins._instances:
+            if plugin.name == name or (
+                self.plugin_type and isinstance(plugin, self.plugin_type)
+            ):
+                return plugin
+        raise ValueError(f"No plugin found with name {name}")
+
+    @contextmanager
+    def plugins(
+        self, *plugins: tuple[str, type[beets.plugins.BeetsPlugin]] | str
+    ):
+        """Context manager to register and load multiple plugins."""
+        self.unload_plugins()
+
+        names = []
+        for plug in plugins:
+            if isinstance(plug, str):
+                names.append(plug)
+            else:
+                names.append(plug[0])
+                self.register_plugin(plug[1], plug[0])
+        self.load_plugins(*names)
+
+        yield
+
+        self.unload_plugins()
+        for name in names:
+            self.unregister_plugin(name)
+
     @contextmanager
     def configure_plugin(self, config: Any):
         self.config[self.plugin].set(config)
-        self.load_plugins(self.plugin)
+        if self.plugin_type is not None:
+            self.register_plugin(self.plugin_type, self.plugin)
+        if self.plugin:
+            self.load_plugins(self.plugin)
 
         yield
 
         self.unload_plugins()
+        if self.plugin_type is not None:
+            self.unregister_plugin(
+                self.plugin
+                or self.plugin_type.__name__.lower().replace("plugin", "")
+            )
+
+
+def get_available_plugins():
+    """Get all available plugins in the beetsplug namespace."""
+    namespace_pkg = importlib.import_module("beetsplug")
+
+    return [
+        m.name
+        for m in pkgutil.iter_modules(namespace_pkg.__path__)
+        if not m.name.startswith("_")
+    ]
+
+
+class PluginTestCase(PluginMixin, TestHelper):
```

Comment:
Let's remove __Case__ from the name to decouple it from `unittest` completely

```suggestion
class PluginTest(PluginMixin, TestHelper):
```